lib/tests/formulae_dependents.rb

```diff
@@ -297,7 +298,7 @@ def install_dependent(dependent, testable_dependents, args:, build_from_source:
            !bottled_on_current_version &&
            !dependent_was_previously_installed &&
            all_tests_passed &&
-           dependent.deps.all? { |d| bottled?(d.to_formula, no_older_versions: true) }
+           required_dependent_deps.all? { |d| bottled?(d.to_formula, no_older_versions: true) }
```

Comment:
Why not
```suggestion
           dependent.effective_deps.all? { |d| bottled?(d.to_formula, no_older_versions: true) }
```
?

Also, perhaps this ship has sailed, but: I prefer changing the semantics of `#deps` so that it actually returns what is currently `#effective_deps`. I can see that this might be a breaking change, but this is also essentially a bug in `#deps` and it is callers who rely on the bugged behaviour who should be fixed. (I doubt there are too many of those in third party taps too.)

Reply:
It's not too late, no.

It's an interesting one, because absolutely nothing in `brew` itself uses `effective_deps`. Test-bot was the opposite it turns out.

It probably makes sense to inverse it (homebrew-bundle uses `formula.deps` for `build_dependencies`, which would probably need adjusting), though I'm not entirely sure how to handle `Dependency.expand` yet (and by extension `recursive_dependencies`).

Reply:
> It's not too late, no.

Yeh, in general with these changes: I'd rather have something that fixes the `install from api` issues land sooner rather than later and a new ideal JSON/API hash format land afterwards which could be completely non-backwards compatible.

Might even be an idea to think about how we can start versioning the JSON format a bit more/better. Perhaps it should be bumped every major Homebrew version?

Reply:
> Might even be an idea to think about how we can start versioning the JSON format a bit more/better. Perhaps it should be bumped every major Homebrew version?

We do need a versioning straategy, yes. As well as policy for what we do with older versions.

I wouldn't say it needs to be bumped every major version. If there are no changes, there's nothing gained from bumping. But every major/minor Homebrew version is a good opportunity to consider a bump.

Reply:
> If there are no changes, there's nothing gained from bumping.

Mmmm, mildly disagree here. If we get consumers of our APIs (be it from formulae.brew.sh or `brew info --json`) used to having to bump the versions they supply to URLs/`brew info` periodically (once a year?) to work around deprecations it'll make it much easier for us to make these sorts of breaking changes in future.

Not saying we should definitely do this but: it's an advantage of bumping even when nothing has changed.